mlrun/runtimes/pod.py

```diff
@@ -247,15 +287,39 @@ def security_context(self, security_context):
             "security_context", security_context
         )
 
-    def to_dict(self, fields=None, exclude=None):
-        exclude = exclude or []
-        _exclude = ["affinity", "tolerations", "security_context"]
-        struct = super().to_dict(fields, exclude=list(set(exclude + _exclude)))
-        api = k8s_client.ApiClient()
-        for field in _exclude:
-            if field not in exclude:
-                struct[field] = api.sanitize_for_serialization(getattr(self, field))
-        return struct
+    def _serialize_field(
+        self, struct: dict, field_name: str = None, strip: bool = False
+    ) -> typing.Any:
+        # we pull the field from self and not from struct because it was excluded from the struct
+        if field_name in self._fields_to_exclude_for_k8s_serialization:
+            k8s_api = k8s_client.ApiClient()
```

Comment:
I feel like we should consider extracting the k8s client to outer scope instead of instantiating it for every field we serialize